[//]: # "File generated from a template. Do not edit this file directly."

# cred-class-field-unobscured-sensitive-input

`typeOptions.password` must be set to `true` in a sensitive node parameter, to obscure the input.

üìã This rule is part of the `plugin:n8n-nodes-base/credentials` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
export class TestApi implements ICredentialType {
	name = "testApi";
	displayName = "Test API";
	documentationUrl = "test";
	properties: INodeProperties[] = [
		{
			displayName: "Password",
			name: "password",
			type: "string",
			default: "",
			required: true,
		},
	];
}

export class TestApi implements ICredentialType {
	name = "testApi";
	displayName = "Test API";
	documentationUrl = "test";
	properties: INodeProperties[] = [
		{
			displayName: "API Key",
			name: "apiKey",
			type: "string",
			default: "",
			required: true,
		},
	];
}

export class TestApi implements ICredentialType {
	name = "testApi";
	displayName = "Test API";
	documentationUrl = "test";
	properties: INodeProperties[] = [
		{
			displayName: "Access Token",
			name: "accessToken",
			type: "string",
			default: "",
			required: true,
		},
	];
}
```

‚úÖ Example of **correct** code:

```js
export class TestApi implements ICredentialType {
	name = "testApi";
	displayName = "Test API";
	documentationUrl = "zammad";
	properties: INodeProperties[] = [
		{
			displayName: "API Key",
			name: "apiKey",
			type: "string",
			typeOptions: { password: true },
			default: "",
			required: true,
		},
	];
}
```

## Links

- [Rule source](../../lib/rules/cred-class-field-unobscured-sensitive-input.ts)
- [Test source](../../tests/cred-class-field-unobscured-sensitive-input.test.ts)
