[//]: # "File generated from a template. Do not edit this file directly."

# node-param-default-wrong-for-options

`default` for an options-type node parameter must be one of the options.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "options",
  default: "wrong",
  options: [
    {
      name: "First",
      value: "first",
    },
    {
      name: "Second",
      value: "second",
    },
  ],
};

const test = {
  displayName: "Test",
  name: "test",
  type: "options",
  default: "wrong",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "options",
  default: "first",
  options: [
    {
      name: "First",
      value: "first",
    },
    {
      name: "Second",
      value: "second",
    },
  ],
};
```

## Links

- [Rule source](../../lib/rules/node-param-default-wrong-for-options.ts)
- [Test source](../../tests/node-param-default-wrong-for-options.test.ts)
