[//]: # "File generated from a template. Do not edit this file directly."

# node-param-placeholder-missing-email

`placeholder` for Email node parameter must exist.

📋 This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
  displayName: "Email",
  name: "email",
  type: "string",
  default: "",
};
```

✅ Example of **correct** code:

```js
const test = {
  displayName: "Email",
  name: "email",
  type: "string",
  default: "",
  placeholder: "hello@n8n.io",
};
```

## Links

- [Rule source](../../lib/rules/node-param-placeholder-missing-email.ts)
- [Test source](../../tests/node-param-placeholder-missing-email.test.ts)
