[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-lowercase-first-char

First char in `displayName` in node parameter or in fixed collection section must be uppercase. Applicable by extension to `name` in options-type or multi-options-type node parameter.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "test",
  name: "test",
  type: "string",
  default: "",
};

const test = {
  name: "user ID",
  value: "userId",
};

const test = {
  displayName: "test",
  name: "test",
  type: "fixedCollection",
  default: "a",
  options: [
    {
      displayName: "Details",
      name: "detailsTest",
      values: [
        {
          displayName: "A",
          name: "a",
          type: "string",
          default: "",
        },
      ],
    },
  ],
};

const test = {
  displayName: "Event",
  name: "event",
  type: "options",
  required: true,
  default: "coupon.created",
  options: [
    {
      name: "coupon.created",
      value: "coupon.created",
    },
  ],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
    displayName: 'Test',
    name: 'test',
    type: 'string',
    default: '',
};

const test = {
    name: 'User ID',
    value: 'userId',
};

const test = {
    displayName: "API Domain",
    name: "apiDomain",
    type: "options",
    options: [
        {
            name: "api.jotform.com",
            value: "api.jotform.com",
        },
        {
            name: "eu-api.jotform.com",
            value: "eu-api.jotform.com",
        },
    ],
    default: "api.jotform.com",
    description:
        'The API domain to use. Use "eu-api.jotform.com" if your account is in based in Europe.',
},

const test = {
    displayName: "SASL Mechanism",
    name: "saslMechanism",
    type: "options",
    options: [
        {
            name: "Plain",
            value: "plain",
        },
        {
            name: "scram-sha-256",
            value: "scram-sha-256",
        },
        {
            name: "scram-sha-512",
            value: "scram-sha-512",
        },
    ],
    default: "plain",
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-lowercase-first-char.ts)
- [Test source](../../tests/node-param-display-name-lowercase-first-char.test.ts)
