[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-nonstandard-for-fixed-collection

`displayName` for top-level fixed collection for create operation must be `Additional Fields`. `displayName` for top-level fixed collection for update operation must be `Update Fields`. `displayName` for top-level fixed collection for get-all operation must be `Options`.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Wrong",
  name: "additionalFields",
  type: "fixedCollection",
  default: {},
  displayOptions: {
    show: {
      resource: ["ticket"],
      operation: ["create"],
    },
  },
};

const test = {
  displayName: "Wrong",
  name: "updateFields",
  type: "fixedCollection",
  default: {},
  displayOptions: {
    show: {
      resource: ["ticket"],
      operation: ["update"],
    },
  },
};

const test = {
  displayName: "Wrong",
  name: "options",
  type: "fixedCollection",
  default: {},
  displayOptions: {
    show: {
      resource: ["ticket"],
      operation: ["getAll"],
    },
  },
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Additional Fields",
  name: "additionalFields",
  type: "fixedCollection",
  default: {},
  displayOptions: {
    show: {
      resource: ["ticket"],
      operation: ["create"],
    },
  },
};

const test = {
  displayName: "Update Fields",
  name: "updateFields",
  type: "fixedCollection",
  default: {},
  displayOptions: {
    show: {
      resource: ["ticket"],
      operation: ["update"],
    },
  },
};

const test = {
  displayName: "Options",
  name: "options",
  type: "fixedCollection",
  default: {},
  displayOptions: {
    show: {
      resource: ["ticket"],
      operation: ["getAll"],
    },
  },
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-nonstandard-for-fixed-collection.ts)
- [Test source](../../tests/node-param-display-name-nonstandard-for-fixed-collection.test.ts)
