[//]: # "File generated from a template. Do not edit this file directly."

# node-param-type-options-missing-from-password

`typeOptions.password` must be set to `true` in a Password node parameter, to obscure the password input.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
	displayName: "Password",
	name: "password",
	type: "string",
	default: "",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
	displayName: "Password",
	name: "password",
	type: "string",
	typeOptions: { password: true },
	default: "",
};
```

## Links

- [Rule source](../../lib/rules/node-param-type-options-missing-from-password.ts)
- [Test source](../../tests/node-param-type-options-missing-from-password.test.ts)
