[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-wrong-for-ignore-ssl-issues

`description` for Ignore SSL node parameter must be `Whether to connect even if SSL certificate validation is not possible`

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Ignore SSL Issues",
  name: "allowUnauthorizedCerts",
  type: "boolean",
  default: true,
  description: "Wrong",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Ignore SSL Issues",
  name: "allowUnauthorizedCerts",
  type: "boolean",
  default: true,
  description:
    "Whether to connect even if SSL certificate validation is not possible",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-wrong-for-ignore-ssl-issues.ts)
- [Test source](../../tests/node-param-description-wrong-for-ignore-ssl-issues.test.ts)
