[//]: # "File generated from a template. Do not edit this file directly."

# node-param-fixed-collection-type-unsorted-items

Items in a fixed-collection-type node parameter must be alphabetized by `displayName` if more than five.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "fixedCollection",
  default: "a",
  options: [
    {
      displayName: "Details",
      name: "details",
      values: [
        {
          displayName: "B",
          name: "b",
          type: "string",
          default: "",
        },
        {
          displayName: "A",
          name: "a",
          type: "string",
          default: "",
        },
        {
          displayName: "C",
          name: "c",
          type: "string",
          default: "",
        },
        {
          displayName: "D",
          name: "d",
          type: "string",
          default: "",
        },
        {
          displayName: "E",
          name: "e",
          type: "string",
          default: "",
          description: 'This is a <a href="https://test.com">link</a>',
        },
      ],
    },
  ],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "fixedCollection",
  default: "a",
  options: [
    {
      displayName: "Details",
      name: "details",
      values: [
        {
          displayName: "A",
          name: "a",
          type: "string",
          default: "",
        },
        {
          displayName: "B",
          name: "b",
          type: "string",
          default: "",
        },
        {
          displayName: "C",
          name: "c",
          type: "string",
          default: "",
        },
        {
          displayName: "D",
          name: "d",
          type: "string",
          default: "",
        },
        {
          displayName: "E",
          name: "e",
          type: "string",
          default: "",
        },
      ],
    },
  ],
};
```

## Links

- [Rule source](../../lib/rules/node-param-fixed-collection-type-unsorted-items.ts)
- [Test source](../../tests/node-param-fixed-collection-type-unsorted-items.test.ts)
