[//]: # "File generated from a template. Do not edit this file directly."

# node-param-default-missing

`default` must be present in a node parameter.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "options",
  options: [
    {
      name: "First Option",
      value: "firstOption",
    },
    {
      name: "Second Option",
      value: "secondOption",
    },
  ],
};

const test = {
  displayName: "Test",
  name: "test",
  type: "options",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "options",
  default: "firstOption",
  options: [
    {
      name: "First Option",
      value: "firstOption",
    },
    {
      name: "Second Option",
      value: "secondOption",
    },
  ],
};

const test = {
  displayName: "Scope",
  name: "scope",
  type: "hidden",
  default: scopes.join(","),
};

const test = {
  displayName: "Auth URI Query Parameters",
  name: "authQueryParameters",
  type: "hidden",
  default: `user_scope=${userScopes.join(" ")}`,
};

const test = {
  displayName: "Value",
  name: "value",
  type: "number",
  default: -1,
};
```

## Links

- [Rule source](../../lib/rules/node-param-default-missing.ts)
- [Test source](../../tests/node-param-default-missing.test.ts)
