[//]: # "File generated from a template. Do not edit this file directly."

# node-param-description-line-break-html-tag

`description` in node parameter must not contain an HTML line break. Applicable by extension to `description` in option in options-type and multi-options-type node parameter.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This is a <br /> sentence",
};

const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This is a <br > sentence",
};

const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This is a <br> sentence",
};

const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This is a </br> sentence",
};

const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This is a </ br> sentence",
};

const test = {
  name: "Test",
  value: "test",
  description: "This is a <br /> sentence",
};

const test = {
  name: "Test",
  value: "test",
  description: "This is a <br/> sentence",
};

const test = {
  name: "Test",
  value: "test",
  description: "This is a <br> sentence",
};

const test = {
  name: "Test",
  value: "test",
  description: "This is a </br> sentence",
};

const test = {
  name: "Test",
  value: "test",
  description: "This is a </br > sentence",
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
  description: "This is a sentence",
};

const test = {
  name: "Test",
  value: "test",
  description: "This is a sentence",
};
```

## Links

- [Rule source](../../lib/rules/node-param-description-line-break-html-tag.ts)
- [Test source](../../tests/node-param-description-line-break-html-tag.test.ts)
