[//]: # "File generated from a template. Do not edit this file directly."

# node-execute-block-wrong-error-thrown

The `execute()` method in a node may only throw `NodeApiError` for failed network requests and `NodeOperationError` for internal errors, not the built-in `Error`.

üìã This rule is part of the `plugin:n8n-nodes-base/non-autofixable` config.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
  execute() {
    throw new Error("An error occurred");
  }
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
  execute() {
    throw new NodeApiError("An error occurred");
  }
}

class TestNode {
  execute() {
    throw new NodeOperationError("An error occurred");
  }
}
```

## Links

- [Rule source](../../lib/rules/node-execute-block-wrong-error-thrown.ts)
- [Test source](../../tests/node-execute-block-wrong-error-thrown.test.ts)
