[//]: # "File generated from a template. Do not edit this file directly."

# node-execute-block-error-missing-item-index

In the `execute()` method in a node, `NodeApiError` and `NodeOperationError` must specify the `itemIndex` as the third argument.

üìã This rule is part of the `plugin:n8n-nodes-base/nodes` config.

## Examples

‚ùå Example of **incorrect** code:

```js
class TestNode {
  execute() {
    throw new NodeApiError(this.getNode(), "An error occurred");
  }
}

class TestNode {
  execute() {
    throw new NodeOperationError(this.getNode(), "An error occurred");
  }
}

class TestNode {
  execute() {
    throw new NodeApiError(this.getNode(), "An error occurred", {
      testIndex: 1,
    });
  }
}
```

‚úÖ Example of **correct** code:

```js
class TestNode {
  execute() {
    throw new NodeApiError(this.getNode(), "An error occurred", {
      itemIndex: i,
    });
  }
}

class TestNode {
  execute() {
    throw new NodeOperationError(this.getNode(), "An error occurred", {
      itemIndex: i,
    });
  }
}
```

## Links

- [Rule source](../../lib/rules/node-execute-block-error-missing-item-index.ts)
- [Test source](../../tests/node-execute-block-error-missing-item-index.test.ts)
