[//]: # "File generated from a template. Do not edit this file directly."

# node-param-display-name-lowercase-first-char

First char in `displayName` in node parameter or in fixed collection section must be uppercase. Applicable by extension to `name` in options-type or multi-options-type node parameter.

📋 This rule is part of the `plugin:n8n-nodes-base/recommended` config.

📋 This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

🔧 Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

❌ Example of **incorrect** code:

```js
const test = {
  displayName: "test",
  name: "test",
  type: "string",
  default: "",
};

const test = {
  name: "user ID",
  value: "userId",
};

const test = {
  displayName: "test",
  name: "test",
  type: "fixedCollection",
  default: "a",
  options: [
    {
      displayName: "Details",
      name: "detailsTest",
      values: [
        {
          displayName: "A",
          name: "a",
          type: "string",
          default: "",
        },
      ],
    },
  ],
};
```

✅ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "string",
  default: "",
};

const test = {
  name: "User ID",
  value: "userId",
};
```

## Links

- [Rule source](../../lib/rules/node-param-display-name-lowercase-first-char.ts)
- [Test source](../../tests/node-param-display-name-lowercase-first-char.test.ts)
