[//]: # "File generated from a template. Do not edit this file directly."

# node-param-collection-type-unsorted-items

Items in collection-type node parameter must be alphabetized by `name` if five or more than five.

üìã This rule is part of the `plugin:n8n-nodes-base/recommended` config.

üìã This rule is part of the `plugin:n8n-nodes-base/autofixable-safe` config.

üîß Run ESLint with `--fix` option to autofix the issue flagged by this rule.

## Examples

‚ùå Example of **incorrect** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "collection",
  default: {},
  options: [
    {
      name: "B",
      value: "b",
    },
    {
      name: "A",
      value: "a",
    },
    {
      name: "C",
      value: "c",
    },
    {
      name: "D",
      value: "d",
    },
    {
      name: "E",
      value: "e",
    },
  ],
};
```

‚úÖ Example of **correct** code:

```js
const test = {
  displayName: "Test",
  name: "test",
  type: "collection",
  default: {},
  options: [
    {
      name: "C",
      value: "c",
    },
    {
      name: "A",
      value: "a",
    },
    {
      name: "B",
      value: "b",
    },
    {
      name: "D",
      value: "d",
    },
  ],
};

const test = {
  displayName: "Test",
  name: "test",
  type: "collection",
  default: {},
  options: [
    {
      name: "A",
      value: "a",
    },
    {
      name: "B",
      value: "b",
    },
    {
      name: "C",
      value: "c",
    },
    {
      name: "D",
      value: "d",
    },
    {
      name: "E",
      value: "e",
    },
  ],
};
```

## Links

- [Rule source](../../lib/rules/node-param-collection-type-unsorted-items.ts)
- [Test source](../../tests/node-param-collection-type-unsorted-items.test.ts)
